src/hotspot/cpu/aarch64/aarch64_neon.ad

```diff
@@ -170,30 +166,32 @@ instruct reinterpretX2D(vecD dst, vecX src)
             n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 16);
   match(Set dst (VectorReinterpret src));
   ins_cost(INSN_COST);
-  format %{ " # reinterpret $dst,$src\t# X2D" %}
+  format %{ " # reinterpret $dst,$src\t# X to D" %}
   ins_encode %{
-    // Resize the vector from 128-bits to 64-bits. The higher 64-bits of
-    // the "dst" register must be cleared to zero.
+    // The higher 64-bits of the "dst" register must be cleared to zero.
     __ orr(as_FloatRegister($dst$$reg), __ T8B,
            as_FloatRegister($src$$reg),
            as_FloatRegister($src$$reg));
   %}
   ins_pipe(vlogical64);
 %}
 
-// ------------------------------ Vector cast -------------------------------
-
-instruct vcvt4Bto4S(vecD dst, vecD src)
+instruct reinterpretS2X(vecX dst, vecD src)
 %{
-  predicate(n->as_Vector()->length() == 4 && n->bottom_type()->is_vect()->element_basic_type() == T_SHORT);
-  match(Set dst (VectorCastB2X src));
-  format %{ "sxtl  $dst, T8H, $src, T8B\t# convert 4B to 4S vector" %}
+  predicate(n->bottom_type()->is_vect()->length_in_bytes() == 16 &&
+            n->in(1)->bottom_type()->is_vect()->length_in_bytes() == 4);
+  match(Set dst (VectorReinterpret src));
+  ins_cost(INSN_COST);
+  format %{ " # reinterpret $dst,$src\t# S2X" %}
   ins_encode %{
-    __ sxtl(as_FloatRegister($dst$$reg), __ T8H, as_FloatRegister($src$$reg), __ T8B);
+    // The upper bits of "src" are expected to have been initialized to zero.
```

Comment:
I think the comment should be:

// The higher bits of the "dst" register must be cleared to zero.

Reply:
Thanks, I'll fix it.